src/Meta/MetaFactory.cpp

```diff
@@ -357,7 +357,50 @@ void MetaFactory::createFromMethod(const clang::ObjCMethodDecl& method, MethodMe
         }
     }
 
-    methodMeta.setFlags(MetaFlags::MethodIsInitializer, method.getMethodFamily() == clang::ObjCMethodFamily::OMF_init); // set MethodIsInitializer flag
+    bool isInitializer = (method.getMethodFamily() == clang::ObjCMethodFamily::OMF_init);
+    methodMeta.setFlags(MetaFlags::MethodIsInitializer, isInitializer); // set MethodIsInitializer flag
+    if (isInitializer) {
+        assert(methodMeta.getSelector().find("init", 0) == 0);
```

Comment:
Correct me if I am wrong but asserts are disabled in our builds, so if this condition is not met you are still going through the next lines.
Also is there a case where you could have an initializer that doesn't start with a "init" prefix. I've experimented a bit but couldn't create one :(